PYME/IO/DataSources/CropDataSource.py

```diff
@@ -29,20 +29,41 @@ def __init__(self,dataSource, xrange=None, yrange=None, trange=None):
         #self.unmixer = unmixer
         self.dataSource = dataSource
         
-        if xrange is None:
+        sizeC = self.dataSource.sizeC
+
+        if xrange is None or tuple(xrange) == (0, -1):
```

Comment:
if you want to use -1 indexing etc ... it would probably be better to use a `slice()` object. `slice.indices` would likely also let you eliminate much of the boilerplate in each of the cases below.

Reply:
but I'm not sure I'm particularly keen on negative indexing in this case.